clientUtils/formatValue.ts

```diff
@@ -70,10 +74,15 @@ export function formatValue(
         if (value !== 0 && Math.abs(value) < targetDigits) {
             if (value < 0) output = `>-${targetDigits}`
             else output = `<${targetDigits}`
-        } else
-            output = d3Format(`${showPlus ? "+" : ""},.${numDecimalPlaces}f`)(
+        } else if (unitIsCurrency) {
+            output = d3Format(unit)(
+                `${showPlus ? "+" : ""}$,.${numDecimalPlaces}f`
```

Comment:
Nice, I wasn't aware that you can pass `$` and it gets replaced with the currency:

```ts
d3.format("($.2f")(-3.5); // localized fixed-point currency, "(£3.50)"
```

But I think this might be causing the problem with omitted units. With number prefixes, we do a recursive call overriding the unit, and the unit gets lost (probably?)

Reply:
I went ahead and made a little sandbox to test the different specifiers: https://observablehq.com/@ikesau/d3-format-interactive-demo

I'm in the process of rewriting the function to rely on d3.format in a much simpler way, with some post-processing at the end for the usecases it can't handle. Hopefully I'll be out on the other side soon!